### PR TITLE
feat(conan): add system_requirements for qtwebengine

### PR DESCRIPTION
**Pull request by ledocc**

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -256,18 +256,28 @@ def requirements(self):
                 self.requires("xkbcommon/0.8.4@bincrafters/stable")
 
     def system_requirements(self):
+        pack_names = []
+
+        if self.options.qtwebengine:
+            if tools.os_info.is_linux:
+                if tools.os_info.with_apt:
+                    pack_names += ["bison", "flex", "gperf", "python",
```
Comment (ericLemanissier) [inline conanfile.py]: Do you think that bison, flex and gperf system package could be replaced with`build_requires` of `bison-installer/3.4.1@bincrafters/stable`, `flex/2.6.4@bincrafters/stable` and `gperf_installer/3.1@conan/stable`  protected by `if not tools.which('foo'):` like https://github.com/bincrafters/conan-qt/blob/testing/5.13.1/conanfile.py#L152 ?

Comment (ledocc) [reply conanfile.py]: Done, I also add test in test_package.
I still have missing dependencies, so don't merge it yet please.

**Comment by Croydon** [comment]
If this is only compatible with Python 2 then I'm not sure if we should get into the trouble of supporting this

Python 2 is EOL at the end of the month and Conan will stop supporting it with 2.0

**Comment by ledocc** [comment]
qtwebengine provided in Qt 5.12.5 require python2 to build, and Qt 5.12 is a LTS release (3 years), out in 2018 december, so even if python 2 is EOL at the end of the month, this is still usefull.

**Comment by ericLemanissier** [comment]
And unfortunately this requirements comes from chromium itself : https://chromium.googlesource.com/chromium/src/+/master/docs/linux_build_instructions.md#system-requirements, so it will not change for qt 5.15, which is LTS and will be released in 2020

File: conanfile.py [hunk 2]
```diff
@@ -573,12 +582,25 @@ def _getenvpath(var):
...
-        with tools.vcvars(self.settings) if self.settings.compiler == "Visual Studio" else tools.no_op():
-            with tools.environment_append({"MAKEFLAGS": "j%d" % tools.cpu_count(), "PKG_CONFIG_PATH": os.getcwd()}):
+
+        build_env = {"MAKEFLAGS": "j%d" % tools.cpu_count(), "PKG_CONFIG_PATH": os.getcwd()}
+        if self.options.qtwebengine:
+            if self.options.python2_bin_dir:
+                build_env['PATH'] = [str(self.options.python2_bin_dir)]
+            if self.settings.compiler in ['gcc', 'clang']:
+                build_env['C_INCLUDE_PATH'] = self.deps_cpp_info['fontconfig'].include_paths
+                build_env['C_PLUS_INCLUDE_PATH'] = self.deps_cpp_info['fontconfig'].include_paths
```
Comment (ericLemanissier) [inline conanfile.py]: it seems to me that you should be using CPLUS_INCLUDE_PATH instead

Comment (ledocc) [reply conanfile.py]: well spot, fixed

**Comment by ericLemanissier** [comment]
merged in https://github.com/bincrafters/conan-qt/commit/45837926490ad7ac7f0c609739f051ad37246f12

**Pull request closed without merging**